src/Service/CoverStore/CloudinaryCoverStoreService.php

```diff
@@ -125,13 +139,23 @@ public function remove(string $folder, string $identifier): void
      *
      * @throws \Cloudinary\Api\Exception\GeneralError
      */
-    public function search(string $folder, string $rawQuery = null): array
+    public function search(string $folder, string $rawQuery = null, bool $useRecursiveSearch = false): array
```

Comment:
I think this could be much cleaner code if we return an `iterator`. Then the `foreach ($result['resources'] as $resources) {` could `yield` items instead of building an array. This would allow us scope `$nextCursorIndex` to this function instead of having it in an array in class scope.